libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -84,88 +85,87 @@ bool es_objects_plugin_impl::index_database( const vector<object_id_type>& ids,
    block_time = db.head_block_time();
    block_number = db.head_block_num();
 
-   // check if we are in replay or in sync and change number of bulk documents accordingly
-   uint32_t limit_documents = 0;
-   if((fc::time_point::now() - block_time) < fc::seconds(30))
-      limit_documents = _es_objects_bulk_sync;
-   else
-      limit_documents = _es_objects_bulk_replay;
-
-   for(auto const& value: ids) {
-      if(value.is<proposal_object>() && _es_objects_proposals) {
-         auto obj = db.find_object(value);
-         auto p = static_cast<const proposal_object*>(obj);
-         if(p != nullptr) {
-            if(action == "delete")
-               remove_from_database(p->id, "proposal");
-            else
-               prepareTemplate<proposal_object>(*p, "proposal");
-         }
-      }
-      else if(value.is<account_object>() && _es_objects_accounts) {
-         auto obj = db.find_object(value);
-         auto a = static_cast<const account_object*>(obj);
-         if(a != nullptr) {
-            if(action == "delete")
-               remove_from_database(a->id, "account");
-            else
-               prepareTemplate<account_object>(*a, "account");
-         }
-      }
-      else if(value.is<asset_object>() && _es_objects_assets) {
-         auto obj = db.find_object(value);
-         auto a = static_cast<const asset_object*>(obj);
-         if(a != nullptr) {
-            if(action == "delete")
-               remove_from_database(a->id, "asset");
-            else
-               prepareTemplate<asset_object>(*a, "asset");
-         }
-      }
-      else if(value.is<account_balance_object>() && _es_objects_balances) {
-         auto obj = db.find_object(value);
-         auto b = static_cast<const account_balance_object*>(obj);
-         if(b != nullptr) {
-            if(action == "delete")
-               remove_from_database(b->id, "balance");
-            else
-               prepareTemplate<account_balance_object>(*b, "balance");
-         }
-      }
-      else if(value.is<limit_order_object>() && _es_objects_limit_orders) {
-         auto obj = db.find_object(value);
-         auto l = static_cast<const limit_order_object*>(obj);
-         if(l != nullptr) {
-            if(action == "delete")
-               remove_from_database(l->id, "limitorder");
-            else
-               prepareTemplate<limit_order_object>(*l, "limitorder");
-         }
-      }
-      else if(value.is<asset_bitasset_data_object>() && _es_objects_asset_bitasset) {
-         auto obj = db.find_object(value);
-         auto ba = static_cast<const asset_bitasset_data_object*>(obj);
-         if(ba != nullptr) {
-            if(action == "delete")
-               remove_from_database(ba->id, "bitasset");
-            else
-               prepareTemplate<asset_bitasset_data_object>(*ba, "bitasset");
+   if(_es_objects_start_es_after_block == 0 || block_number > _es_objects_start_es_after_block) {
```

Comment:
Can remove the == 0 check here, since if it is 0 then block_number will always be greater than that.